### PR TITLE
tutorial: automated-publishing-container-builder

### PR DESCRIPTION
**Pull request by ahmetb**
Adding a new tutorial on how to automatically publish static websites to GCS using Google Cloud Container Builder.

Currently the images point to `./path`, let me know if they need to be updated to GCS locations as other articles do. I could not find any guidance about the image URLs in the [Style Guide](https://cloud.google.com/community/tutorials/styleguide#images). Keeping them as relative URLs for now so that reviewing would be easier on GitHub.

### CODE REVIEW

**Comment by jmdobry** [comment]
> I could not find any guidance about the image URLs in the Style Guide.

Created #102 to track. I copied your images to the Cloud Storage bucket and updated the links.

**Comment by ahmetb** [comment]
🙌  thanks so much!

**Comment by jmdobry** [comment]
I'll still leave it to @hegemonic to review the actual tutorial content.

File: tutorials/automated-publishing-container-builder/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+![Specify main page](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/configure-website.png)
+
+Now, configure your domain name’s DNS records to [create a CNAME
+record][gcs-hosting] that points to the Google Cloud Storage. This makes clients
+requesting your website point to the Google Storage APIs.
+
+## Set up automated builds
+
+We will use [Google Cloud Container Builder][gcb] and the [Build Triggers][bt]
+feature to make sure we upload our website automatically every time we push a
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/we/you/ (both places)

File: tutorials/automated-publishing-container-builder/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+new git commit to our source repository.
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/our/your/

File: tutorials/automated-publishing-container-builder/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+above, pick it here:
+
+![Select the Git repository](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/select-repo.png)
+
+In the next screen
+
+- give this Build Trigger a name (e.g. publish-website)
+- choose Build Configuration "cloudbuild.yaml"
+- choose Trigger Type ”Branch”
+- type cloudbuild.yaml to the file location
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: replace with "- set the file location to /cloudbuild.yaml"

File: tutorials/automated-publishing-container-builder/index.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+![Create build trigger](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/create-trigger.png)
+
+Now, create a `cloudbuild.yaml` file with the following contents on your
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/on/in/

File: tutorials/automated-publishing-container-builder/index.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+repository. Note that you can add files to your repository on GitHub website, or
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/GitHub/GitHub's/

File: tutorials/automated-publishing-container-builder/index.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+by cloning the repository on your development machine:
+
+```yaml
+steps:
+  - name: gcr.io/cloud-builders/gsutil
+    args: ["-m", "rsync", "-r", "-c", "-d", "./vision/explore-api", "gs://hello.alp.im"]
+```
+
+This YAML file declares a build step with `gsutil -m rsync` command and makes
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/with/with the/

File: tutorials/automated-publishing-container-builder/index.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+sure that the website is uploaded to the storage bucket. The `-m` flag
+accelerates upload by processing multiple files in parallel and the `-c` flag
+avoids re-uploading unchanged files.
+
+The last command in the `args` is the name of your storage bucket prefixed with
+`gs://`.
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: Add: "Be sure to change this argument to the correct value."

File: tutorials/automated-publishing-container-builder/index.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+In our sample repository, we wanted to upload `vision/explore-api/` directory,
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: Move this paragraph before the previous paragraph, and replace the beginning with: "If you're using the sample repository, you should upload the directory `./vision/explore-api`. If you would like..."

File: tutorials/automated-publishing-container-builder/index.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+if you would like to upload your entire repository to the storage bucket, make
+sure to change the value to `.` in the YAML file.
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/the value/this value/

File: tutorials/automated-publishing-container-builder/index.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+After saving the file, commit and push the changes:
+
+    git add cloudbuild.yaml
+    git commit -m 'Add build configuration'
+    git push
+
+### Trigger the first build
+
+Once you push the `cloudbuild.yaml` file to your repository and create the Build
+Trigger, we can now kick off the first build manually. Head over to the Google
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/we can now/you can/

File: tutorials/automated-publishing-container-builder/index.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+Cloud Console [Build Triggers][p6n-triggers] section, click “Run Trigger” and
+choose the the branch (i.e. master) to build.
+
+![Trigger the first build manually](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/trigger-build.png)
+
+Now click the “Build history” on the left and watch the build job execute and
+succeeed:
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/succeeed/succeed/

File: tutorials/automated-publishing-container-builder/index.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+![Build history shows the executing or completed builds](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/build-history.png)
+
+Remember that after now, every commit pushed to any branch of your GitHub
+repository will trigger a new build and publish contents to your website. If you
+need to change which git branches or tags you use for publishing, you can update
+the Build Trigger configuration.
+
+## Try it out
+
+Point your browser to your website URL and see if it works
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/works/works:/

File: tutorials/automated-publishing-container-builder/index.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+![See if your website works](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/browser.png)
+
+## Clean up
+
+After you no longer need the artifacts of this tutorial, you can clean up the
+following resources on the Google Cloud Console to prevent incurring:
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/incurring:/incurring additional charges:/

File: tutorials/automated-publishing-container-builder/index.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+- Storage: delete bucket named as your website
+- Container Registry &rarr; Build Triggers: delete build trigger.
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: remove period

File: tutorials/automated-publishing-container-builder/index.md [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+- Development &rarr; Repositories: delete mirrored repository.
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: remove period

File: tutorials/automated-publishing-container-builder/index.md [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+
+## Before you begin
+
+1. Make sure you have a custom domain name (e.g. example.com).
+1. Make sure the source code for your static website is hosted on a GitHub or
+   BitBucket repository.
+1. Make sure you verified ownership of your domain on [Google Webmaster
+   Central][gwc]. (Do not include `http://` or `https://` in the URL for the
+   purposes of this demo.)
+1. Make sure you have a Project on Google Cloud Console to host your website.
```
Comment (ahmetb) [inline tutorials/automated-publishing-container-builder/index.md]: @hegemonic I have mixed feelings:

> image

> image

but I will change.

File: tutorials/automated-publishing-container-builder/index.md [hunk 17]
```diff
@@ -23,77 +23,88 @@ Builder][gcb].
...
+1. Make sure you have a Project on Google Cloud Platform Console to host your
+   website.
 
 ## Set up a storage bucket
 
-By uploading your website contents as files to Google Cloud Storage, you can
-[host your static website][gcs-hosting] on buckets. First, we need to create a
-bucket. Head to [Storage][p6n-storage] section of Google Cloud Console and type
-in your domain name (e.g. `www.example.com`) and create the bucket:
+By uploading your website contents as files to Google Cloud Platform Storage,
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/Platform//

File: tutorials/automated-publishing-container-builder/index.md [hunk 18]
```diff
@@ -23,77 +23,88 @@ Builder][gcb].
...
+![Select the Git
+repository](https://storage.googleapis.com/gcp-community/tutorials/automated-publishing-container-builder/select-repo.png)
 
 In the next screen
 
 - give this Build Trigger a name (e.g. publish-website)
 - choose Build Configuration "cloudbuild.yaml"
 - choose Trigger Type ”Branch”
-- type cloudbuild.yaml to the file location
+- Set the fiel location to `cloudbuild.yaml`
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/fiel/file/

File: tutorials/automated-publishing-container-builder/index.md [hunk 19]
```diff
@@ -102,17 +113,18 @@ steps:
...
-This YAML file declares a build step with `gsutil -m rsync` command and makes
-sure that the website is uploaded to the storage bucket. The `-m` flag
+This YAML file declares a build step with the `gsutil -m rsync` command and
+makes sure that the website is uploaded to the storage bucket. The `-m` flag
 accelerates upload by processing multiple files in parallel and the `-c` flag
 avoids re-uploading unchanged files.
 
-The last command in the `args` is the name of your storage bucket prefixed with
-`gs://`.
+If you are using the sample repository, you should upload the directory
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: s/directory//

File: tutorials/automated-publishing-container-builder/index.md [hunk 20]
```diff
@@ -102,17 +113,18 @@ steps:
...
+`vision/explore-api/` directory. If you would like to upload your entire
```
Comment (hegemonic) [inline tutorials/automated-publishing-container-builder/index.md]: Change directory to `./vision/explore-api` to match the actual command.

**Comment by ahmetb** [comment]
Thanks @hegemonic. I addressed the comments.

**Pull request merged**